cime_config/machines/config_machines.xml

```diff
@@ -597,15 +597,21 @@
     <BATCH_SYSTEM>slurm</BATCH_SYSTEM>
     <SUPPORTED_BY>e3sm</SUPPORTED_BY>
     <MAX_TASKS_PER_NODE>56</MAX_TASKS_PER_NODE>
+    <MAX_TASKS_PER_NODE compiler="gnugpu">16</MAX_TASKS_PER_NODE>
+    <MAX_TASKS_PER_NODE compiler="crayclanggpu">16</MAX_TASKS_PER_NODE>
+    <MAX_TASKS_PER_NODE compiler="amdclanggpu">16</MAX_TASKS_PER_NODE>
```

Comment:
We would need to accommodate cases to potentially use all available CPU cores for threading. So, we should let these go up to 56 (low-noise mode) default.